server/sonar-server/src/test/java/org/sonar/elasticsearch/test/EsTestCluster.java

```diff
@@ -0,0 +1,969 @@
+  }
+
+  /**
+   * Returns the number of nodes in the cluster.
+   */
+  public synchronized int size() {
+    return this.nodes.size();
+  }
+
+  private synchronized void publishNode(NodeAndClient nodeAndClient) {
+    assert !nodeAndClient.node().isClosed();
+    nodes.put(nodeAndClient.name, nodeAndClient);
+  }
+
+
+  /**
+   * Returns an {@link Iterable} over all clients in this test cluster
+   */
+  public synchronized Iterable<Client> getClients() {
+    ensureOpen();
+    return () -> {
+      ensureOpen();
+      final Iterator<NodeAndClient> iterator = nodes.values().iterator();
+      return new Iterator<Client>() {
+
+        @Override
+        public boolean hasNext() {
+          return iterator.hasNext();
+        }
+
+        @Override
+        public Client next() {
+          return iterator.next().client(random);
+        }
+
+        @Override
+        public void remove() {
+          throw new UnsupportedOperationException("");
+        }
+
+      };
+    };
+  }
+
+  /**
+   * Ensures that any breaker statistics are reset to 0.
+   *
+   * The implementation is specific to the test cluster, because the act of
+   * checking some breaker stats can increase them.
+   */
+  private void ensureEstimatedStats() {
+    if (size() > 0) {
+      // Checks that the breakers have been reset without incurring a
+      // network request, because a network request can increment one
+      // of the breakers
+      for (NodeAndClient nodeAndClient : nodes.values()) {
+        final IndicesFieldDataCache fdCache = getInstanceFromNode(IndicesService.class, nodeAndClient.node).getIndicesFieldDataCache();
+        // Clean up the cache, ensuring that entries' listeners have been called
+        fdCache.getCache().refresh();
+
+        final String name = nodeAndClient.name;
+        final CircuitBreakerService breakerService = getInstanceFromNode(CircuitBreakerService.class, nodeAndClient.node);
+        CircuitBreaker fdBreaker = breakerService.getBreaker(CircuitBreaker.FIELDDATA);
+        assertThat("Fielddata breaker not reset to 0 on node: " + name, fdBreaker.getUsed(), equalTo(0L));
+        // Anything that uses transport or HTTP can increase the
+        // request breaker (because they use bigarrays), because of
+        // that the breaker can sometimes be incremented from ping
+        // requests from other clusters because Jenkins is running
+        // multiple ES testing jobs in parallel on the same machine.
+        // To combat this we check whether the breaker has reached 0
+        // in an assertBusy loop, so it will try for 10 seconds and
+        // fail if it never reached 0
+        try {
+          assertBusy(new Runnable() {
+            @Override
+            public void run() {
+              CircuitBreaker reqBreaker = breakerService.getBreaker(CircuitBreaker.REQUEST);
+              assertThat("Request breaker not reset to 0 on node: " + name, reqBreaker.getUsed(), equalTo(0L));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this assertion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2186)